zipline/lib/adjustment.pyx

```diff
@@ -530,9 +533,62 @@ cdef class Datetime64Overwrite(Datetime64Adjustment):
     """
     cpdef mutate(self, int64_t[:, :] data):
         cdef Py_ssize_t row, col
+        cdef int64_t value = self.value
 
         # last_col + 1 because last_col should also be affected.
         for col in range(self.first_col, self.last_col + 1):
             # last_row + 1 because last_row should also be affected.
             for row in range(self.first_row, self.last_row + 1):
-                data[row, col] = self.value
+                data[row, col] = value
+
+
+cdef class _ObjectAdjustment(Adjustment):
+    """
+    Base class for adjustments that operate on arbitrary objects.
+
+    We use only this for categorical data, where our data buffer is an array of
+    indices into an array of unique Python string objects.
+    """
+    cdef:
+        readonly object value
+
+    def __init__(self,
+                 Py_ssize_t first_row,
+                 Py_ssize_t last_row,
+                 Py_ssize_t first_col,
+                 Py_ssize_t last_col,
+                 object value):
+        super(Adjustment, self).__init__(
```

Comment:
Yes.  I think we're being saved here by the fact that Cython implicitly calls all your parent's `__new__`s before yours.  Will update.